--- Python-2.7.1/setup.py	2011/07/10 23:56:34	1.1
+++ Python-2.7.1/setup.py	2011/07/10 23:57:47
@@ -1454,14 +1454,14 @@
 
 
         # Platform-specific libraries
-        if platform == 'linux2':
+        if (platform in ('linux2', 'linux3')):
             # Linux-specific modules
             exts.append( Extension('linuxaudiodev', ['linuxaudiodev.c']) )
         else:
             missing.append('linuxaudiodev')
 
-        if (platform in ('linux2', 'freebsd4', 'freebsd5', 'freebsd6',
-                        'freebsd7', 'freebsd8')
+        if (platform in ('linux2', 'linux3', 'freebsd4', 'freebsd5',
+                        'freebsd6', 'freebsd7', 'freebsd8')
             or platform.startswith("gnukfreebsd")):
             exts.append( Extension('ossaudiodev', ['ossaudiodev.c']) )
         else:
